src/components/core/geoChart/GeoChartRenderer.tsx

```diff
@@ -112,20 +116,38 @@ export default class GeoChartRenderer extends React.PureComponent<IGeoChartRende
             this.chart.removeLayer("settlement-label");
         }
 
-        const geoData = getGeoData(buckets, executionResponse.dimensions);
-        chart.addSource(DEFAULT_DATA_SOURCE_NAME, createPushpinDataSource(executionResult, geoData));
-        chart.addLayer(
-            createPushpinDataLayer(DEFAULT_DATA_SOURCE_NAME, executionResult, geoData, selectedSegmentItem),
-            "state-label", // pushpin will be rendered under state/county label
-        );
+        this.geoData = getGeoData(buckets, executionResponse.dimensions);
+
+        chart.addSource(DEFAULT_DATA_SOURCE_NAME, createPushpinDataSource(executionResult, this.geoData));
```

Comment:
`executionResult` & `executionResponse` should be pre-proceed in `CoreGeoChart` component.
We would like to reuse some pre-processed data for legend and avoid duplication.

Reply:
This fix does not relate to Legend logic so refactoring should be done in Legend impl.
How do you think?

Reply:
it's ok